troposphere/static/js/components/modals/instance/InstanceLaunchWizardModal.react.js

```diff
@@ -479,30 +349,52 @@ export default React.createClass({
     },
 
     renderBasicOptions: function() {
+        let provider = this.state.provider;
+        let imageVersion = this.state.imageVersion;
+        let projectList = stores.ProjectStore.getAll() || null;
+        let sizes = stores.SizeStore.getAll() || null;
+        let identities = stores.IdentityStore.getAll() || null;
+        let bootScriptList = stores.ScriptStore.getAll();
 
-        return (
-            <BasicLaunchStep {...this.state.basicLaunch}
-                advancedIsDisabled={this.state.advancedIsDisabled}
-                launchIsDisabled={this.state.launchIsDisabled}
-                image={this.state.image}
-                onNameChange={this.onNameChange}
-                onVersionChange={this.onVersionChange}
-                onProjectChange={this.onProjectChange}
-                onProviderChange={this.onProviderChange}
-                onSizeChange={this.onSizeChange}
-                onRequestResources={this.onRequestResources}
-                viewAdvanced={this.viewAdvanced}
-                onCancel={this.onCancelModal}
-                onSubmitLaunch={this.onSubmitLaunch}
-                onBack={this.onBack}
-            />
-        );
+        var imageVersionList;
+        if (this.state.image) {
+            imageVersionList = stores.ImageVersionStore.fetchWhere({image_id: this.state.image.id});
+        }
+
+        var providerList, providerSizeList, resourcesUsed; 
+        if (provider && imageVersion) {
+            providerList = new Backbone.Collection(imageVersion.get('machines').map((item) => item.provider));
+            providerSizeList = stores.SizeStore.fetchWhere({
+                provider__id: provider.id
+            });
+            resourcesUsed = stores.InstanceStore.getTotalResources(provider.id);
+        }
+
+        return React.createElement(BasicLaunchStep, _.extend({}, this.state, {
+            projectList,
+            providerList,
+            providerSizeList,
+            resourcesUsed,
+            imageVersionList,
+            onNameChange: this.onNameChange,
+            onVersionChange: this.onVersionChange,
+            onProjectChange: this.onProjectChange,
+            onProviderChange: this.onProviderChange,
+            onSizeChange: this.onSizeChange,
+            onRequestResources: this.onRequestResources,
+            viewAdvanced: this.viewAdvanced,
+            onCancel: this.onCancelModal,
+            onSubmitLaunch: this.onSubmitLaunch,
+            onBack: this.onBack,
+        }));
     },
```

Comment:
I vote this should remain JSX with the spread. Mostly because it is visually consistent with other components and it's a method React provides for this purpose that a new developer will recognize.